src/permissions.rs

```diff
@@ -0,0 +1,175 @@
+use codec::{Decode, Encode};
+use srml_support::dispatch;
+
+use crate::constraint::*;
+use crate::principles::*;
+
+/// Permissions for an instance of a Class in the versioned store.
+#[derive(Encode, Decode, Default, Eq, PartialEq, Clone, Debug)]
+pub struct ClassPermissions<ClassId, AccountId, GroupId, PropertyIndex, BlockNumber>
+where
+    ClassId: Ord,
+    AccountId: Ord + Clone,
+    GroupId: Ord + Clone,
+    PropertyIndex: Ord,
+{
+    // concrete permissions
+    /// Permissions that are applied to entities of this class, defines who in addition to
+    /// root origin can update and delete entities of this class.
```

Comment:
We cannot delete entities yet. (Will we be able to do this in the future? Not decided yet)
```suggestion
    /// root origin can update entities of this class.
```